src/main/java/dk/kb/datahandler/util/HarvestTimeUtil.java

```diff
@@ -26,43 +40,181 @@
  */
 
 public class HarvestTimeUtil {
-    
-    public static final String defaultStartDate="1900-01-01T00:00:00Z"; //Used as start if no file has been written yet for that target
+
+    public static final String DEFAULT_START_DATE="1900-01-01T00:00:00Z"; //Used as start if no file has been written yet for that target
     private static final Logger log = LoggerFactory.getLogger(HarvestTimeUtil.class);
-    private static final Charset UTF8= Charset.forName("UTF-8");    
-    
+    private static final Charset UTF8= StandardCharsets.UTF_8;         
+    private static final String DAY_PATTERN = "yyyy-MM-dd";
+
+
     public static synchronized String loadLastHarvestTime(OaiTargetDto oaiTarget) throws Exception{            
         String oaiTargetNameFile = getFileNameFromOaiTarget(oaiTarget);        
         return loadLastHarvestTime(oaiTargetNameFile);        
     }
-    
-    
+
+
     public static synchronized void updateDatestampForOaiTarget(OaiTargetDto oaiTarget, String datestamp) throws Exception{        
-        
-        if (!validateDataFormat(datestamp)) {
+
+        //Hack to fix datestamp returned from Preservica. Format returned are not in OAI standard and is parsed wrong (downgrade to seconds) when given to preservica.
+        //Only preservica 6 does not, preservica 5 gives correct format.
+        if (datestamp.length() >20) { // 2021-03-24T19:57:34.123Z -> 2021-03-24T19:57:34Z 
+            datestamp=datestamp.substring(0,19)+"Z";
+        }
+        else if(datestamp.length()==10) {             
+            datestamp=datestamp+"T00:00:00Z";
+        }
+
+        if (!validateOaiDateFormat(datestamp)) {
             log.error("Datestamp not valid format:"+datestamp +" for Oai target:"+oaiTarget.getName());
             throw new InvalidArgumentServiceException("Datastamp not valid format:" + datestamp);
         }
-        
-        
+
+
         String oaiTargetNameFile = getFileNameFromOaiTarget(oaiTarget);
         updateDatestampForOaiTarget(oaiTargetNameFile, datestamp);        
     }
-    
-    
+
+    /**
+     * Validate day format is of form at yyyy-MM-dd.
+     * Example: 2024-02-09
+     * 
+     * @param day  Validate is of correct format
+     */
+    public static boolean validateDayFormat(String day) {
+        SimpleDateFormat simpleDateFormat = new SimpleDateFormat(DAY_PATTERN,Locale.ROOT);
+        simpleDateFormat.setLenient(false); //day must exist
+        try {    	
+            Date date = simpleDateFormat.parse(day);    	   
+        }
+        catch(Exception e) {
+            log.warn("Parsing of day format failed:"+day);
+            return false;
+        }
+        return true;			
+
+    }
+
+    /**
+     * 
+     * From a java date format yyyy-MM-dd
+     * 
+     * @param date  Java date object
+     */
+    public static String formatDate2Day(Date date) {
```

Comment:
This method is only used for unit testing. Please move it to the test classes.

Reply:
fixed.